File: src/devmatching/Solution1.java
package devmatching;
import java.util.*;

public class Solution1 {
    public int solution(String p, String s) {
        int answer = 0;
        char[] eachP= p.toCharArray();
        char[] eachS= s.toCharArray();
        for (int i = 0; i < eachP.length; i++) {
            int password = Integer.parseInt( Character.toString(eachP[i]));
            int find = Integer.parseInt( Character.toString(eachS[i]));
            int a = (Math.min(password,find) + 10) - Math.max(password,find);
            int b = Math.abs(password-find);
            answer += Math.min(a,b);
        }
        return answer;
    }
}

File: src/섬머코딩2020/Solution1.kt
package 섬머코딩2020

class Solution1 {
    fun solution(p: Int): Int {
        var answer = 0
        for (i in p+1 .. 10000) {
            if (check(i)) {
                answer = i
                break
            }
        }
        return answer
    }
    private fun check(p: Int) : Boolean {
        val map = HashMap<Char,Boolean>()
        val arr = p.toString().toCharArray()
        for (i in arr.indices) {
            if (!map.containsKey(arr[i])) map[arr[i]] = true
            else return false
        }
        return true
    }
}